src/core/network/qgsrangerequestcache.cpp

```diff
@@ -0,0 +1,152 @@
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsrangerequestcache.h"
+
+#include <QtDebug>
+#include <QFile>
+#include <QDir>
+#include <QDateTime>
+
+QgsRangeRequestCache::QgsRangeRequestCache()
+{
+
+}
+
+QByteArray QgsRangeRequestCache::entry( const QNetworkRequest &request )
+{
+  QString hash = rangeFileName( request );
+  QByteArray arr = readFile( hash );
+  return arr;
+}
+
+bool QgsRangeRequestCache::hasEntry( const QNetworkRequest &request )
+{
+  QDir dir( mCacheDir );
+  return dir.exists( rangeFileName( request ) );
+}
+
+void QgsRangeRequestCache::registerEntry( const QNetworkRequest &request, QByteArray data )
+{
+  QString hash = rangeFileName( request );
+  writeFile( hash, data );
+  expire();
+}
+
+void QgsRangeRequestCache::clear()
+{
+  QDir dir( mCacheDir );
+  for ( QFileInfo info : dir.entryInfoList() )
+  {
+    removeFile( info.filePath() );
+  }
+}
+
+void QgsRangeRequestCache::setCacheDirectory( const QString &path )
+{
+  QString cachePath = path;
+  if ( !cachePath.endsWith( QDir::separator() ) )
+  {
+    cachePath.push_back( QDir::separator() );
+  }
+  mCacheDir = cachePath;
+}
+
+void QgsRangeRequestCache::setCacheSize( qint64 cacheSize )
+{
+  mMaxDataSize = cacheSize;
+  expire();
+}
+
+QByteArray QgsRangeRequestCache::readFile( const QString &fileName )
+{
+  QFile file( fileName );
+  file.open( QFile::OpenModeFlag::ReadOnly );
+  if ( !file.isOpen() || !file.isReadable() )
+    return QByteArray();
+  return file.readAll();
+}
+
+void QgsRangeRequestCache::writeFile( const QString &fileName, QByteArray data )
+{
+  QFile file( fileName );
+  file.open( QFile::OpenModeFlag::WriteOnly );
+  if ( !file.isOpen() || !file.isWritable() )
+    return;
+  file.write( data );
+  file.close();
+}
+
+void QgsRangeRequestCache::removeFile( const QString &fileName )
+{
+  if ( fileName.isEmpty() )
+    return;
+  QFile::remove( fileName );
+}
+
+QFileInfoList QgsRangeRequestCache::cacheFiles()
+{
+  QDir dir( mCacheDir );
+  QFileInfoList filesList = dir.entryInfoList( QDir::Filter::Files );
+  std::sort( filesList.begin(), filesList.end(), []( QFileInfo & f1, QFileInfo & f2 )
+  {
+    QDateTime t1 = f1.fileTime( QFile::FileTime::FileAccessTime );
+    QDateTime t2 = f2.fileTime( QFile::FileTime::FileAccessTime );
+    return t1 > t2;
+  } );
+  return filesList;
+}
+
+void QgsRangeRequestCache::expire()
+{
+  QDir dir( mCacheDir );
+  QFileInfoList filesList = cacheFiles();
+  qint64 totalSize = 0;
+  for ( QFileInfo info : filesList )
+  {
+    totalSize += info.size();
+  }
+  while ( totalSize > mMaxDataSize )
+  {
+    QFileInfo info = filesList.back();
+    filesList.pop_back();
+    totalSize -= info.size();
+    removeFile( info.filePath() );
+  }
+}
+
+QStringList QgsRangeRequestCache::cacheEntries()
```

Comment:
do we need both cacheFiles() and cacheEntries() functions?

Reply:
Ah yes the 2 do the same thing, I removed cacheFiles function.